inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -65,19 +65,19 @@ def formdata_to_model(obj, formdata):
     data['collections'] = [{'primary': "HEP"}]
     if form_fields['type_of_doc'] == 'thesis':
         data['collections'].append({'primary': "THESIS"})
-    if "field_categories" in data:
+
+    if get_value(form_fields, "arxiv_eprints.categories", None):
```

Comment:
This is the change that makes the unit tests fail: we populate `external_system_numbers` from `arxiv_id` only if the form contained `arxiv_eprints.categories`.